### ISSUE TITLE
Segarrays complete within locale files (like string arrays)

### ISSUE THREAD

**Issue by 21771** [Bears-R-Us/arkouda#2387]
**Feature Description**
Currently datasets cannot be read from individual locale files if the dataset is a segarray.  It would be nice if segarrays written to multiple files could be implemented in a self-contained manner like strings. 

There may be occasions when it's not practical to dedicate enough resources to read a particularly large file if only a small number of records are needed. It would be preferable to be able to only load the needed locales (or worst case, chunk through the entire data set -- but only loading fractions of the entire file at one time.)

**Example Behavior**
If applicable, please provide a code block showing expected behavior of the feature.

**Describe Workarounds**
Is there currently a way to achieve the desired functionality (even if it is not ideal)? If so, please provide it for context.

**Additional context**
Add any other context or screenshots about the feature request here. If the requested feature is based on another library (i.e. numpy, pandas, scipy), please provide a link to their supporting documentation for reference.

**Comment by Ethan-DeBandi99** [comment]
@21771 - I will have to look into this. SegArray should be set up to store the same way that Strings does. So that all values from a segment are in the same file. I will do some testing and see if I can replicate what you are seeing. Is this happening in Parquet, HDF5 or both?

**Comment by 21771** [comment]
Happening in hdf5, didn't try in parquet. It successfully reads in LOCALE0 alone, but it complains on other files because segments don't start at zero.

**Comment by Ethan-DeBandi99** [comment]
@21771 - OK I see what happened. Previously, we stored SegArrays as 2 independent pdarrays that were named in a way that we could identify the values and the offsets. When doing this, the locality of the segments was not maintained. The work in PR #2128 updated SegArray to save under an HDF5 group like Strings and uses the same concept of saving the complete segment on a locale. Running the code below in the latest tag, `v2023.04.07` works as expected. Looking at when the change was merged, it should have made the `v2023.03.01` release. However, I know there were some issues with the implementation that have since been resolved. Again, I verified in the current release, but older files will not be able to take advantage of this due to the way we previously saved SegArray. Also, SegArray read/write support for Parquet was added in PR #2187. This would not be available until `v2023.03.24`, but it follows the same storage convention as HDF5 now does and functions the same as the code below if you replace the respective HDF5 function with their Parquet equivalents. I provided my test case and the output below for reference.

**Code**
```python
import arkouda as ak

ak.connect()

s = ak.segarray(ak.arange(0, 500, 5), ak.arange(500))#ak.randint(0, 20, 500))

fpath = "/path/to/file/segarray_test"

s.to_hdf(fpath)

print(ak.read_hdf([fpath+"_LOCALE0001", fpath+"_LOCALE0003"]))

ak.disconnect()
```

**Output**
```python
SegArray([
[125 126 127 128 129]
[130 131 132 133 134]
[135 136 137 138 139]
...
[485 486 487 488 489]
[490 491 492 493 494]
[495 496 497 498 499]
])
```
